### PR TITLE
tracing: readme cleanup

### PR DESCRIPTION
**Pull request by hawkw**
This branch cleans up the `tracing` readme, including fixing
links that incorrectly pointed to `tracing-core`, and adds a 
link to `tracing-timing` in related crates.

Signed-off-by: Eliza Weisman <eliza@buoyant.io>

### CODE REVIEW

File: tracing/README.md [hunk 1]
```diff
@@ -241,23 +244,42 @@ tracing::subscriber::with_default(my_subscriber, || {
...
 ```
 
-This approach allows trace data to be collected by multiple subscribers within
-different contexts in the program. Note that the override only applies to the
-currently executing thread; other threads will not see the change from with_default.
+Any trace events generated outside the context of a subscriber will not be collected.
 
-Any trace events generated outside the context of a
-subscriber will not be collected.
+The executable itself may use the `tracing` crate to instrument itself as well.
```
Comment (jonhoo) [inline tracing/README.md]: This sentence is hard to parse.

File: tracing/README.md [hunk 2]
```diff
@@ -241,23 +244,42 @@ tracing::subscriber::with_default(my_subscriber, || {
...
 
-The executable itself may use the `tracing` crate to instrument itself as
-well.
+In addition to `tracing` and `tracing-core`, the [`tokio-rs/tracing`] repository
+several additional crates designed to be used with the `tracing` ecosystem.
```
Comment (jonhoo) [inline tracing/README.md]: Missing word.

File: tracing/README.md [hunk 3]
```diff
@@ -241,23 +244,42 @@ tracing::subscriber::with_default(my_subscriber, || {
...
+
+In particular, the following crates are likely to be of interest:
+
+ - [`tracing-futures`] provides a compatibility layer with the `futures`
+   crate, allowing spans to be attached to `Future`s, `Stream`s, and `Executor`s.
+ - [`tracing-fmt`] provides a `Subscriber` implementation for
+   logging formatted trace data to stdout, with similar filtering and
+   formatting to the `env-logger` crate.
+ - [`tracing-log`] provides a compatibility layer with the `log` crate,
+   allowing log `Record`s to be recorded as `tracing` `Event`s within the
```
Comment (jonhoo) [inline tracing/README.md]: I think the reference to "log `Record`s" is probably unhelpful here, as most people (I think) aren't used to think of `log` in terms of the `Record` type. Instead, I suspect _most_ people only think of log in terms of its logging macros.

File: tracing/src/lib.rs [hunk 4]
```diff
@@ -292,8 +292,7 @@
 //! currently executing thread; other threads will not see the change from with_default.
 //! with that subscriber as the default.
 //!
-//! Any trace events generated outside the
-//! context of a subscriber will not be collected.
+//! Any trace events generated outside the context of a subscriber will not be collected.
 //!
 //! The executable itself may use the `tracing` crate to instrument itself
```
Comment (jonhoo) [inline tracing/src/lib.rs]: Same comment here that this phrasing is somewhat confusing.

File: tracing/README.md (hunk 2 above)
Comment (jonhoo) [reply tracing/README.md]: The same is true in the `src/lib.rs` that this was copied from.

**Comment by hawkw** [comment]
@jonhoo thanks for the notes! I've made all your suggested changes in 8a3a38c4412b365a156a6813dba266abdda74bce

**Comment by jonhoo** [comment]
LGTM :+1:
